### PR TITLE
Code cleanup: take advantage of C# 8 features.

### PR DESCRIPTION
**Pull request by simeoncran**
Also adds some more #nullable enable annotation so quieten code analysis.

None of these changes should have changed any behaviors.

### CODE REVIEW

**Comment by derobinsMSFT** [comment]
> // Licensed to the .NET Foundation under one or more agreements.

If all of these error suppressions are gone, is this file needed anymore? Or does Code Analysis always expect to find this file even if it's "empty"?

---
Refers to: GlobalSuppressions.cs:1 in 1c4bdab. [](commit_id = 1c4bdab0a9aa129eeb1254306534c8273eadda43, deletion_comment = False)

File: source/YamlData/YamlWriter.cs [hunk 1]
```diff
@@ -192,14 +192,10 @@ bool TryInlineObject(YamlObject obj, int maximumWidth, [MaybeNullWhen(false)] ou
             {
                 switch (obj.Kind)
```
Comment (derobinsMSFT) [inline source/YamlData/YamlWriter.cs:193]: >switch (obj.Kind) [](start = 16, length = 17)

`return (obj.Kind) switch`?

**Review by derobinsMSFT — APPROVED** [review]
:shipit:

File: source/YamlData/YamlWriter.cs (hunk 1 above)
Comment (simeoncran) [reply source/YamlData/YamlWriter.cs:193]: I tried, and failed. I think it's hitting a limitation of the nullable analysis. Even though the TryInline methods all return a string with MaybeNullWhen(false), the nullable analysis complains if this is a switch expression.

---
In reply to: [508131986](https://github.com/windows-toolkit/Lottie-Windows/pull/370#discussion_r508131986) [](ancestors = 508131986)

**Pull request merged**